test/libsolidity/smtCheckerTests/bmc_coverage/msg_value_4.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+contract A {
+    uint x = msg.value;
+    constructor() {
+        assert(x == 0); // should hold
+    }
+}
```

Comment:
@blishko you did add some already, like this one for example

Reply:
There is no inheritance there, `A` and `B` are independent.
That just tests that with a payable constructor, `msg.value` can be anything, but with a non-payable constructor, it must be `0`. In the latter case, I needed to introduce state variable, because the compiler does not even allow to access `msg.value` if the constructor is not payable.

Reply:
Ah true

Reply:
I have added tests with inheritance, `special/msg_value_inheritance_1.sol` and `special/msg_value_inheritance_2.sol`.